packages/accordion/src/index.js

```diff
@@ -0,0 +1,406 @@
+  },
+  forwardedRef
+) {
+  /*
+   * You shouldn't switch between controlled/uncontrolled. We'll check for a
+   * controlled component and track any changes in a ref to show a warning.
+   */
+  const wasControlled = typeof controlledIndex !== "undefined";
+  const { current: isControlled } = useRef(wasControlled);
+
+  const id = useId(props.id);
+
+  const [activeIndex, setActiveIndex] = useState(
+    isControlled
+      ? controlledIndex
+      : defaultIndex != null
+      ? defaultIndex
+      : toggle
+      ? -1
+      : 0
+  );
+
+  /*
+   * We will store all AccordionTrigger refs inside this array to manage focus.
+   */
+  const focusabledTriggerNodes = useRef([]);
+
+  /*
+   * Loop through children and find all `disabled` items. This will allow us
+   * to determine next/previous focusable items. We also need to count children
+   * with the `active` prop to deal with instances where `allowMultiple` is set
+   * on an accordion with multiple active children.
+   */
+  const childrenArray = useMemo(() => {
+    const arr = React.Children.toArray(children);
+    return Array.isArray(arr) ? arr : [];
+  }, [children]);
+
+  const enabledIndices = useMemo(() => {
+    let enabledIndices = [];
+
+    for (let i = 0; i < childrenArray.length; i++) {
+      const child = childrenArray[i];
+      if (!(typeof child.type === "string" || child.props.disabled === true)) {
+        enabledIndices.push(i);
+      }
+    }
+    return enabledIndices;
+  }, [childrenArray]);
+
+  if (__DEV__) {
+    warning(
+      !((isControlled && !wasControlled) || (!isControlled && wasControlled)),
+      "Accordion is changing from controlled to uncontrolled. Accordion should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled Accordion for the lifetime of the component. Check the `index` prop being passed in."
+    );
+  }
+
+  function onSelectPanel(index) {
+    onChange && onChange(index);
+    /*
+     * Before updating the active item internally, check that:
+     *   - Component is uncontrolled
+     *   - If toggle is not allowed, check that the change isn't coming from an
+     *     item that is already active.
+     */
+    if (!isControlled && !(activeIndex === index && !toggle)) {
+      setActiveIndex(activeIndex === index && toggle ? -1 : index);
+    }
+  }
+
+  const context = {
+    accordionId: id,
+    activeIndex,
+    focusabledTriggerNodes,
+    enabledIndices,
+    onSelectPanel: readOnly ? noop : onSelectPanel,
+    readOnly
+  };
+
+  if (isControlled && controlledIndex !== activeIndex) {
+    /*
+     * If the component is controlled, we'll sync internal state with the
+     * controlled state
+     */
+    setActiveIndex(controlledIndex);
+  }
+
+  return (
+    <AccordionContext.Provider value={context}>
+      <div data-reach-accordion="" ref={forwardedRef} {...props}>
+        {Children.map(children, (child, _index) =>
+          cloneElement(child, { _index })
```

Comment:
If we are cloning any reason to use context?

Reply:
We are using context values in nested components and it seemed simpler than passing those props all the way down the tree.